Src/FluentAssertions/Equivalency/EqualityStrategyCache.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using FluentAssertions.Common;
+using JetBrains.Annotations;
+
+namespace FluentAssertions.Equivalency;
+
+internal sealed class EqualityStrategyCache
+{
+    private readonly List<Type> referenceTypes = new();
+    private readonly List<Type> valueTypes = new();
+    private readonly ConcurrentDictionary<Type, EqualityStrategy> typeCache = new();
+
+    [CanBeNull]
+    private readonly Func<Type, EqualityStrategy> defaultStrategy;
+
+    private bool? compareRecordsByValue;
+
+    public bool? CompareRecordsByValue
+    {
+        get => compareRecordsByValue;
+        set
+        {
+            compareRecordsByValue = value;
+            typeCache.Clear();
+        }
+    }
+
+    public EqualityStrategyCache([CanBeNull] Func<Type, EqualityStrategy> defaultStrategy)
```

Comment:
🙃 I prefer the constructor to precede any properties (like [AV2406](https://csharpcodingguidelines.com/layout-guidelines/#AV2406) states)

Reply:
Fixed in [commit 8249eea](https://github.com/fluentassertions/fluentassertions/pull/2431/commits/8249eea1fb65ce69c8ecd533ddca4ffb437488c3).